include/sway/lock.h

```diff
@@ -0,0 +1,23 @@
+#ifndef _SWAY_LOCK_H
+#define _SWAY_LOCK_H
+#include <wlr/types/wlr_session_lock_v1.h>
+
+enum sway_lock_state {
+	SESSIONLOCK_STATE_UNLOCKED,
+	SESSIONLOCK_STATE_LOCKED,
+	SESSIONLOCK_STATE_ABANDONED,
+};
+
+struct sway_session_lock_manager {
+	enum sway_lock_state state;
```

Comment:
Maybe we could replace `state` with a `bool locked` field? Then we can just check `manager->lock == NULL` to figure out whether the locker has crashed. This ensures we only have a single source of truth.